File: MG/App/Frontend/WEB_flutter/lib/models/commentReaction.dart
class CommentReaction {
  int id;
  int commentID;
  int userID;
  int type;
  DateTime time;
  bool isRead;

  CommentReaction(commentID, userID, type, DateTime time, bool isRead) {
    this.commentID = commentID;
    this.userID = userID;
    this.type = type;
    this.time = time;
    this.isRead = isRead;
  }

  Map<String, dynamic> toMap() {
    var map = Map<String, dynamic>();
    if (id != null) 
      map['id'] = id;
    map['commentID'] = commentID;
    map['userID'] = userID;
    map['type'] = type;
    map['time'] = time.toIso8601String();
    map['isRead'] = isRead;
    return map;
  }

  CommentReaction.fromObject(dynamic o) {
    this.id = o['id'];
    this.commentID = o['commentID'];
    this.userID = o['userID'];
    this.type = o['type'];
    this.time = DateTime.tryParse(o['time']);
    this.isRead = o['isRead'];
  }
}

File: MG/App/Backend/mojGradApp/mojGradApp/Models/CommentReaction.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace mojGradApp.Models
{
    public class CommentReaction
    {
        public int Id { get; set; }
        public int CommentID { get; set; }
        public int UserID { get; set; }
        public int Type { get; set; }
        public DateTime Time { get; set; }
        public bool IsRead { get; set; }
        public bool IsApproved { get; set; }

        public virtual Comments Comment { get; set; }
        public virtual UserDatas User { get; set; }
    }
}